eq-author/src/components/routing/RoutingDestinationContentPicker.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { isEmpty, get } from "lodash";
+
+import ContentPickerSelect from "components/ContentPickerSelect";
+import { DESTINATION } from "components/ContentPickerSelect/content-types";
+
+import AvailableRoutingDestinationsQuery from "components/routing/AvailableRoutingDestinationsQuery";
+
+const getDisplayName = (selected, data) => {
+  if (selected.hasOwnProperty("absoluteDestination")) {
+    return selected.absoluteDestination.displayName;
+  } else {
+    if (selected.logicalDestination === "EndOfQuestionnaire") {
+      return "End of questionnaire";
+    }
+
+    if (isEmpty(data)) {
+      return;
+    }
+
+    if (data.questionPages.length) {
+      return data.questionPages[0].displayName;
+    } else if (data.sections.length) {
+      return data.sections[0].displayName;
+    }
+  }
+};
+
+export const UnwrappedRoutingDestinationContentPicker = ({
+  data,
+  path,
+  selected,
+  ...otherProps
+}) => {
+  const destinationData = get(data, path);
+  return (
+    <ContentPickerSelect
+      name="routingDestination"
+      contentTypes={[DESTINATION]}
+      destinationData={destinationData}
+      selectedContentDisplayName={getDisplayName(selected, destinationData)}
```

Comment:
It feels like this logic of managing the selected item including extracting the displayName should be handled by the `ContentPickerSelect`

Reply:
My assumption was that it provided a consistent output onChange. So I was just expecting that input. And as it has knowledge of that format already it wouldn't be different.

Reply:
Maybe this is something we can look at later.